pandas/core/arrays/categorical.py

```diff
@@ -1390,17 +1390,24 @@ def check_for_ordered(self, op):
                             "you can use .as_ordered() to change the "
                             "Categorical to an ordered one\n".format(op=op))
 
-    def argsort(self, ascending=True, kind='quicksort', *args, **kwargs):
-        """
-        Returns the indices that would sort the Categorical instance if
-        'sort_values' was called. This function is implemented to provide
-        compatibility with numpy ndarray objects.
+    def _values_for_argsort(self):
+        return self._codes.copy()
 
-        While an ordering is applied to the category values, arg-sorting
-        in this context refers more to organizing and grouping together
-        based on matching category values. Thus, this function can be
-        called on an unordered Categorical instance unlike the functions
-        'Categorical.min' and 'Categorical.max'.
+    def argsort(self, *args, **kwargs):
```

Comment:
Not sure if this changes our opinion on `_values_for_argsort`, but the apparently Python2 has issues with passing through the arguments correctly to the `super()` call.

```
____________________ TestCategoricalSort.test_numpy_argsort ____________________

self = <pandas.tests.categorical.test_sorting.TestCategoricalSort object at 0x7efcb391f950>

    def test_numpy_argsort(self):
        c = Categorical([5, 3, 1, 4, 2], ordered=True)
    
        expected = np.array([2, 4, 1, 3, 0])
>       tm.assert_numpy_array_equal(np.argsort(c), expected,
                                    check_dtype=False)

pandas/tests/categorical/test_sorting.py:26: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../miniconda3/envs/pandas/lib/python2.7/site-packages/numpy/core/fromnumeric.py:886: in argsort
    return argsort(axis, kind, order)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = [5, 3, 1, 4, 2]
Categories (5, int64): [1 < 2 < 3 < 4 < 5]
ascending = -1, kind = 'quicksort', args = (None,), kwargs = {}

    def argsort(self, ascending=True, kind='quicksort', *args, **kwargs):
        """
            Returns the indices that would sort the Categorical instance if
            'sort_values' was called. This function is implemented to provide
            compatibility with numpy ndarray objects.
    
            While an ordering is applied to the category values, arg-sorting
            in this context refers more to organizing and grouping together
            based on matching category values. Thus, this function can be
            called on an unordered Categorical instance unlike the functions
            'Categorical.min' and 'Categorical.max'.
    
            Returns
            -------
            argsorted : numpy array
    
            See also
            --------
            numpy.ndarray.argsort
            """
        # Keep the implementation here just for the docstring.
        return super(Categorical, self).argsort(ascending=ascending, kind=kind,
>                                               *args, **kwargs)
E       TypeError: argsort() got multiple values for keyword argument 'ascending'
```

Changing the Categorical.argsort to accept just `*args, **kwargs` fixes things, since `ExtensionArray` does the argument validation, but it's a bit unfortunate.